### DISCUSSION TITLE
testnet genesis and seeds are not working

### DISCUSSION THREAD

**Discussion by wdcs-sohamjadiya** [okx/exchain#2939] [General]
Hello guys,

I am trying to setup a okex chain testnet node. but while setting up seeds and genesis for okex they are not working as expected I am facing an error, kindly help on the same.

**Comment by giskook** [comment]
Hi @wdcs-SohamJadiya ,
> I am trying to setup a okex chain testnet node. but while setting up seeds and genesis for okex they are not working as expected I am facing an error, kindly help on the same.


OKC testnet only support start from snapshot. you can refer to [this article](https://forum.okt.club/d/32-how-to-join-the-public-testnet). and download the snapshot [here](https://static.okex.org/cdn/oec/snapshot/index.html).
Hope helps.

**Reply by wdcs-sohamjadiya** [reply]
Hello @giskook,

I've tried the same extracted the testnet-s0-20230129-18086780-rocksdb.tar. and started the binary again. but facing below error while its trying to dial seed.
`E[2023-01-30|07:48:27.626][130930] Error dialing seed. module=p2p err="dial tcp 35.176.111.229:36656: i/o timeout" seed=d7eec05e6449945c8e0fd080d58977d671eae588@35.176.111.229:36656`

please help me to resolve this issue.

**Reply by giskook** [reply]
I had met this problem too,
1. I wait for the connect for a while, 
2. I removed addrbook.json which is located in ${EXCHAIND_PATH}/config folder then restart OKC. but It may complain too.

BTW, can you send me the start command?

**Reply by wdcs-sohamjadiya** [reply]
Hello 

I am using this start command: `exchaind start --home ${EXCHAIND_PATH}`

**Reply by giskook** [reply]
Okay, the command is correct. 
Have you tried to delete the addrbook.json and restart?
BTW, could you execute `exchaind version --long | grep -E "version|commit"` and paste the output here?

**Reply by wdcs-sohamjadiya** [reply]
Output from the above command, I am trying with removing the addrbook.json and restarting it again.
```
version: v1.6.7.2
commit: 9834981c4448365a0a347e07c0ef71063f4f675e
go: go version go1.19.5 linux/amd64

```

**Reply by wdcs-sohamjadiya** [reply]
same issue even after removing the addrbook.json.
```
E[2023-01-30|11:32:02.567][133250] Error dialing seed. module=p2p err="dial tcp 35.176.111.229:36656: i/o timeout" seed=d7eec05e6449945c8e0fd080d58977d671eae588@35.176.111.229:36656
E[2023-01-30|11:32:02.618][133250] Error dialing seed. module=p2p err="dial tcp 3.13.150.20:36656: connect: connection refused" seed=b7c6bdfe0c3a6c1c68d6d6849f1b60f566e189dd@3.13.150.20:36656
E[2023-01-30|11:32:03.619][133250] Error dialing seed. module=p2p err="dial tcp 18.162.106.25:36656: i/o timeout" seed=223b5b41d1dba9057401def49b456630e1ab2599@18.162.106.25:36656
E[2023-01-30|11:32:03.619][133250] Couldn't connect to any seeds. module=p2p 
```

**Reply by wdcs-sohamjadiya** [reply]
can you please share the seed addresses?

**Reply by giskook** [reply]
Hi @wdcs-SohamJadiya,
1. Please use v1.6.8.2
2. When compile exchaind seed is set by default.
3. Try below configure.
```
--p2p.persistent_peers=c39ca38c650b920f9b6c5a9aed7ff904124ec3ad@176.34.14.79:26656,d937e21fd489809add23dc3e55ed78d947217aa8@35.73.6.150:26656,a3eb3c129e49137d5e1665bbf87b6f2be70a0b85@35.72.118.200:26656,b171a9ef83b95c28182bc7aa7ea8639d04e572e7@3.8.247.238:26656,3a700a3849c401396b1c51eb65b1cfc1a8c4394b@3.11.150.42:26656,0208e66d4ca746ec535a0bf05409dc87df408b15@18.180.118.97:26656,ed1819fa1eae52ddec4c0f8cddd80b9cb7c68a22@54.238.76.5:26656,0b3ab9597a66f2f94c8efa4ccb6ed2a1f44d4184@175.41.248.12:26656,67b29551c7c3839ad6c93379991344266aec3829@18.178.135.201:36656,cd07b20b596aac923a1d5bb022581e279755aff1@18.178.138.1:26656,6ce06a89a968a4204d9dcb470f2275767c8dfa68@52.192.2.29:26656,6dd38d96df3ccbca95769ee15bdfdd952ad007c5@52.198.159.49:26656,fcc95bfee6ea74bdf385be3a29072329603676e5@54.150.103.141:26656,7b5b3041d2b3546a236b6df7ff7e06a19a5cae46@54.249.243.203:26656,c098585e299ff7afe6f354c4431550d6919bdd0d@18.177.169.67:26656,5b44fb4af4cfb72286162cb49a3bc04cb8187775@35.72.103.57:26656,358e3399b68fb67787f1386c685db2e75352d9eb@18.178.126.121:36656,0de948586fb30293d1dd14a99ebc3f719deb7c6f@35.162.198.68:26656,284e87518752c8f655fe217113fa86ba7d6ca72f@52.27.74.158:36656,7f2b8a6b9b8b12247e6992aeb32d69e169c2f5ac@52.36.197.107:36656
```

**Reply by wdcs-sohamjadiya** [reply]
okay let me try that.

**Reply by wdcs-sohamjadiya** [reply]
Hello @giskook, what minimum size instance would be required for the testnet node?

**Reply by wdcs-sohamjadiya** [reply]
I am using GCP compute engines machines for the cloud setup. can you please help me with the right size?
I've tried with 8 vcpus and 16 GB RAM, but process is getting killed

**Reply by giskook** [reply]
We run testnet on AWS with 4 cpus / 32GB RAM / IOPS 16000 
1. I sugguest you configure the exchaind with configure belows

```
--rocksdb.opts max_open_files=10000
--iavl-cache-size=1000000
--iavl-fast-storage-cache-size=1000000
--commit-gap-height=100
```

2. with configure above, If exchaind still OOM, lower the value.

BTW exchaind can use tcmalloc as memory allocator, OKC team provides a command to install tcmalloc.

1. cd exchain
2. run `make tcmalloc`
3. `make testnet OKCMALLOC=tcmalloc`

**Reply by wdcs-sohamjadiya** [reply]
Okay understood, and Thank you very much it worked. and testnode got completely synced as well. If I'll face any issue with the memory allocation I'll use the tcmalloc flags to use it as memory allocator.
